src/app/features/user/delete-account/delete-account.component.spec.ts

```diff
@@ -0,0 +1,168 @@
+        it(
+            'should not attempt to delete account with an invalid password',
+            fakeAsync(() => {
+                const userServiceSpy = fixture.debugElement.injector.get(UserService);
+                const deleteAccountSpy = spyOn(userServiceSpy, 'deleteAccount').and.callThrough();
+
+                component.f.password.setValue('123');
+
+                component.onSubmit();
+                tick();
+                expect(component.submitted).toBe(true, 'Submitted status should be true');
+
+                expect(deleteAccountSpy.calls.any()).toBe(false, 'UserService.deleteAccount should not be called');
+
+                expect(component.loading).toBe(false, 'Loading status should be false');
+                expect(component.submitted).toBe(true, 'Submitted status should be true');
+
+            })
+        );
+
+
+        it(
+            'Should display success message when attempt is successful',
+            fakeAsync(() => {
+                const userServiceSpy = fixture.debugElement.injector.get(UserService);
+                const alertServiceSpy = fixture.debugElement.injector.get(AlertService);
+                const alertSuccessSpy = spyOn(alertServiceSpy, 'success').and.callThrough();
+
+                const deleteAccountSpy = spyOn(userServiceSpy, 'deleteAccount').and.callThrough();
+
+                component.f.password.setValue('123vsrer%fdf');
+
+                component.onSubmit();
+                tick();
+                expect(component.submitted).toBe(true, 'Submitted status should be true');
+                expect(deleteAccountSpy.calls.any()).toBe(true, 'UserService.deleteAccount should be called');
+                expect(alertSuccessSpy.calls.any()).toBe(true, 'Alert success service called');
+                flush()
+
+            })
+        );
+
+        it(
+            'Should display an error when the password is wrong',
+            fakeAsync(() => {
+                const userServiceSpy = fixture.debugElement.injector.get(UserService);
+                const alertServiceSpy = fixture.debugElement.injector.get(AlertService);
+                const alertErrorSpy = spyOn(alertServiceSpy, 'error').and.callThrough();
+
+                const deleteAccountSpy = spyOn(userServiceSpy, 'deleteAccount').and.returnValue(
+                    throwError('UserService test failure')
+                );
+
+                component.f.password.setValue('wrongP@sswo3d');
+
+                component.onSubmit();
+                tick();
+                expect(component.submitted).toBe(true, 'Submitted status should be true');
+
+                expect(deleteAccountSpy.calls.any()).toBe(true, 'UserService.deleteAccount should be called');
+                expect(alertErrorSpy.calls.any()).toBe(true, 'Alert error service called');
+
+                expect(component.loading).toBe(false, 'Loading status should be true');
+                    flush();
+            })
+        );
+
+    });
+
```

Comment:
please also add the other test cases to make sure that the api is called / not with correct cases